lib/chewy.rb

```diff
@@ -65,8 +65,20 @@ class << self
     # A thread-local variables accessor
     # @return [Hash]
     def current
-      Thread.current[:chewy] ||= {}
+      initialize_current_storage
+
+      current
+    end
+
+    # @api private
+    def initialize_current_storage
+      Thread.current.thread_variable_set(:chewy, {})
+
+      define_singleton_method(:current) do
```

Comment:
I think it would be less complex with `def current` doing it all without  dynamic overriding method after first use.
https://www.rubydoc.info/stdlib/core/Thread#thread_variable%3F-instance_method to check if variable is set and set default if it's not would be fine.

Reply:
C'mon, it's just a few lines of code and a pretty common trick. Are there any drawbacks of such an approach?

Reply:
I think there is even other issue - in case of two threads first one will call `thread_variable_set`,
but all other threads will already use overriden method that only `_get`.
This way in multi-threaded environment access to `Chewy.current` from other thread than first one would end up with:
```
irb(main):002:0> Thread.current.thread_variable_get(:chewy)
=> nil
```

(as Thread.current is relative to the caller; it would be issue-less only in case of Thread.main)

Reply:
Good point @kml